core/src/main/java/com/linecorp/armeria/common/http/DefaultHttpParameters.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.http;
+
+import static io.netty.util.HashingStrategy.JAVA_HASHER;
+
+import io.netty.handler.codec.DefaultHeaders;
+import io.netty.handler.codec.ValueConverter;
+import io.netty.util.HashingStrategy;
+
+/**
+ * Default implementation of {@link HttpParameters} which uses the {@link HashingStrategy#JAVA_HASHER}
+ * to support case-sensitive parameter names.
+ */
+public class DefaultHttpParameters
+        extends DefaultHeaders<String, String, HttpParameters> implements HttpParameters {
+
+    /**
+     * Creates a new instance with a default value converter.
+     */
+    @SuppressWarnings("unchecked")
+    public DefaultHttpParameters() {
+        super(JAVA_HASHER, StringValueConverter.INSTANCE);
+    }
+
+    /**
+     * Create a new instance.
+     * @param valueConverter Used to convert values to/from native types.
+     */
+    @SuppressWarnings("unchecked")
+    public DefaultHttpParameters(ValueConverter<String> valueConverter) {
```

Comment:
- How about hiding/removing the constructors that accept `ValueConverter` and/or `NameValidator` until we are sure a user wants to customize them? (because there's no reason to expose something a user may not be interested)
- How about adding a constructor parameter: `DefaultHttpParameters(int arraySizeHint)`?